### PR TITLE
Slither warning review

### PR DESCRIPTION
**Pull request by prat-gpt**
- All other warnings still being shown by Slither after this PR do not apply in our contracts.

### CODE REVIEW

File: src/base/PoolInfoUtils.sol [hunk 1]
```diff
@@ -2,11 +2,23 @@
...
-import './interfaces/IPool.sol';
-
-import '../libraries/external/Auctions.sol';
-import '../libraries/Buckets.sol';
-import '../libraries/external/PoolCommons.sol';
+import { IPool, IERC20Token } from 'src/base/interfaces/IPool.sol';
+
+import { Auctions } from 'src/libraries/external/Auctions.sol';
+import { Buckets } from 'src/libraries/Buckets.sol';
+import { PoolCommons } from 'src/libraries/external/PoolCommons.sol';
```
Comment (MikeHathaway) [inline src/base/PoolInfoUtils.sol]: Nit: from alignment

File: src/base/Pool.sol [hunk 2]
```diff
@@ -8,19 +8,20 @@ import '@openzeppelin/contracts/utils/Multicall.sol';
 import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import { IERC20 }      from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
```
Comment (MikeHathaway) [inline src/base/Pool.sol]: Can also explicit import the remaining open zeppelin libraries

File: src/base/PositionManager.sol [hunk 3]
```diff
@@ -10,23 +10,18 @@ import '@openzeppelin/contracts/utils/Multicall.sol';
 import '@openzeppelin/contracts/security/ReentrancyGuard.sol';
 import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
```
Comment (MikeHathaway) [inline src/base/PositionManager.sol]: Oz explicit imports

File: src/erc721/ERC721Pool.sol [hunk 4]
```diff
@@ -2,10 +2,10 @@
 
 pragma solidity 0.8.14;
 
-import './interfaces/IERC721Pool.sol';
-import './interfaces/IERC721Taker.sol';
-import '../base/FlashloanablePool.sol';
-import './interfaces/IERC721NonStandard.sol';
+import 'src/erc721/interfaces/IERC721Pool.sol';
+import 'src/erc721/interfaces/IERC721Taker.sol';
```
Comment (MikeHathaway) [inline src/erc721/ERC721Pool.sol]: Any reason explicit importing not used here?

**Review by MikeHathaway — COMMENTED** [review]
LGTM just had a few styling comments

File: src/erc721/ERC721Pool.sol (hunk 4 above)
Comment (prat-gpt) [reply src/erc721/ERC721Pool.sol]: pr was a work in progress, forgot to mark it. Explicit imports are being used for this as well now.

File: src/base/Pool.sol [hunk 5]
```diff
@@ -2,25 +2,48 @@
...
+    AuctionsState,
+    DepositsState,
+    LoansState,
+    InflatorState,
+    InterestState,
+    PoolBalancesState,
+    ReserveAuctionState,
+    Bucket,
+    BurnEvent,
+    Liquidation
```
Comment (EdNoepel) [inline src/base/Pool.sol:24]: Do we want to alphabetize these?

File: src/base/Pool.sol [hunk 6]
```diff
@@ -51,7 +74,7 @@ abstract contract Pool is Clone, ReentrancyGuard, Multicall, IPool {
 
     mapping(uint256 => Bucket) internal buckets;   // deposit index -> bucket
 
-    uint256 internal poolInitializations;
+    bool internal poolInitializations;
```
Comment (EdNoepel) [inline src/base/Pool.sol]: Recommend renaming the variable, since it's no longer (and never really was) a count.

**Review by grandizzy — COMMENTED** [review]
this looks really great! there's though some warnings re dead code, can you check what it takes to import only what's needed here? https://github.com/ajna-finance/contracts/actions/runs/3845361248/jobs/6549421242#step:6:6

**Review by EdNoepel — COMMENTED** [review]
A few comments inline, but looks good.

**Comment by prat-gpt** [comment]
> this looks really great! there's though some warnings re dead code, can you check what it takes to import only what's needed here? https://github.com/ajna-finance/contracts/actions/runs/3845361248/jobs/6549421242#step:6:6

Slither is claiming all PoolHelper methods to be as unused. It is not able to recognise the concept of free functions.

File: src/base/Pool.sol (hunk 5 above)
Comment (prat-gpt) [reply src/base/Pool.sol:24]: Alphabetize would be better. Would update.

**Comment by grandizzy** [comment]
> > this looks really great! there's though some warnings re dead code, can you check what it takes to import only what's needed here? https://github.com/ajna-finance/contracts/actions/runs/3845361248/jobs/6549421242#step:6:6
> 
> Slither is claiming all PoolHelper methods to be as unused. It is not able to recognise the concept of free functions.

kk, I think we should be good with this lefover, we can look into solution after audit starts (like maybe importing only needed functions?)

**Comment by prat-gpt** [comment]
> kk, I think we should be good with this lefover, we can look into solution after audit starts (like maybe importing only needed functions?)

we are doing exactly same in this pr, only used functions are being imported. But u are right, we can revisit this after audit starts and verify

File: src/base/Pool.sol (hunk 5 above)
Comment (prat-gpt) [reply src/base/Pool.sol:24]: [d89d769](https://github.com/ajna-finance/contracts/pull/487/commits/d89d769a7a2ccb94992bd60c2e03f3ea15a2b5f6) updated here

File: src/base/Pool.sol (hunk 6 above)
Comment (prat-gpt) [reply src/base/Pool.sol]: updated [67cbaba](https://github.com/ajna-finance/contracts/pull/487/commits/67cbaba39f504912d47951f43100248822e9b940)

**Review by EdNoepel — APPROVED** [review]
I'm good with this now.

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**